### PR TITLE
build: use moov-io/base/log

### PR DESCRIPTION
**Pull request by atonks2**
Using base/log changes the format of the logs. The new application logs will look like this:
```
ts=2020-11-12T18:54:04Z msg="Starting moov-io/imagecashletter server version v0.5.0-dev2" level=info package=main
ts=2020-11-12T18:54:04Z msg="admin server listening on [::]:9093" package=main level=info
ts=2020-11-12T18:54:04Z msg="binding to :8083 for HTTP server" level=info package=main
ts=2020-11-12T18:58:51Z msg="created file=4d2b5737b24d54eaf25b96c533e4071b1c37420e" level=info package=main
ts=2020-11-12T18:59:18Z msg="created file=717142737e1cb028d32009c68c725b7717b612c4" package=main level=info requestID=abc
ts=2020-11-12T18:59:18Z method=POST path=/files/create package=main level=info requestID=abc status=201 duration=185.083µs
ts=2020-11-12T18:59:29Z msg="validated file" package=main level=info fileID=717142737e1cb028d32009c68c725b7717b612c4
ts=2020-11-12T18:59:44Z msg="rendering file contents" package=main fileID=717142737e1cb028d32009c68c725b7717b612c4 level=info
ts=2020-11-12T18:59:44Z method=GET path=/files/717142737e1cb028d32009c68c725b7717b612c4/contents status=200 duration=174.861µs level=info package=main
ts=2020-11-12T18:59:48Z msg="rendering file" level=info package=main 
ts=2020-11-12T18:59:48Z duration=27.374µs level=info package=main method=GET path=/files/717142737e1cb028d32009c68c725b7717b612c4 status=200
ts=2020-11-12T18:59:51Z msg="rendering file" level=info package=main 
ts=2020-11-12T18:59:51Z package=main duration=26.929µs method=GET path=/files/abc status=200 level=info
ts=2020-11-12T19:00:00Z msg=interrupt level=info package=main errored=true
ts=2020-11-12T19:00:00Z msg="http server closed" level=info package=main
ts=2020-11-12T19:00:01Z msg="admin server closed" package=main level=info```

### CODE REVIEW

File: cmd/server/files.go [hunk 1]
```diff
@@ -60,16 +60,20 @@ func getCashLetterId(w http.ResponseWriter, r *http.Request) string {
 
 func getFiles(logger log.Logger, repo ICLFileRepository) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		if requestID := moovhttp.GetRequestID(r); requestID != "" {
+			logger.Set("requestID", requestID)
```
Comment (vxio) [inline cmd/server/files.go:64]: on calls to `Set` or `With`, we have to redeclare the logger: `logger = logger.Set()`

File: cmd/server/files.go [hunk 2]
```diff
@@ -60,16 +60,20 @@ func getCashLetterId(w http.ResponseWriter, r *http.Request) string {
...
+		}
+
 		w = wrapResponseWriter(logger, w, r)
 
 		files, err := repo.getFiles() // TODO(adam): implement soft and hard limits
 		if err != nil {
-			logger.Log("files", fmt.Sprintf("error getting ICL files: %v", err), "requestID", moovhttp.GetRequestID(r))
-			moovhttp.Problem(w, err)
+			logged := logger.LogErrorf("error getting ICL files: %v", err)
+			moovhttp.Problem(w, logged.Err())
```
Comment (vxio) [inline cmd/server/files.go:72]: I recommend the following:
```go
err = logger.LogErrorf("error getting ICL files: %v, err).Err()
moovhttp.Problem(w, err)
```

File: cmd/server/files.go [hunk 3]
```diff
@@ -60,16 +60,20 @@ func getCashLetterId(w http.ResponseWriter, r *http.Request) string {
...
 			return
 		}
 		requestID := moovhttp.GetRequestID(r)
-		logger.Log("files", fmt.Sprintf("found %d files", len(files)), "requestID", requestID)
+		logger.Logf("found %d files requestID=%s", len(files), requestID)
```
Comment (vxio) [inline cmd/server/files.go:76]: I don't think you need to add `requestID` here since you set it above.

File: cmd/server/files.go [hunk 4]
```diff
@@ -80,6 +84,10 @@ func getFiles(logger log.Logger, repo ICLFileRepository) http.HandlerFunc {
 
 func createFile(logger log.Logger, repo ICLFileRepository) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		if requestID := moovhttp.GetRequestID(r); requestID != "" {
+			logger.Set("requestID", requestID)
```
Comment (vxio) [inline cmd/server/files.go:88]: Same comment here about reinitializing the logger

**Review by vxio — COMMENTED** [review]
I made a few comments that applies to a couple of different places in the code.

File: cmd/server/files.go (hunk 1 above)
Comment (atonks2) [reply cmd/server/files.go:64]: Ohhh that's good to know. Thanks for pointing it out.

File: cmd/server/files.go (hunk 3 above)
Comment (atonks2) [reply cmd/server/files.go:76]: Oh you're right. I thought I caught all of those.

**Review by vxio — APPROVED** [review]

**Comment by atonks2** [comment]
Thanks for reviewing @vxio!

**Review by adamdecaf — APPROVED** [review]

**Pull request merged**